tests/test_bs.py

```diff
@@ -388,6 +390,156 @@ def test_transaction(setup_network):
 
 
+def test_transactions(setup_network):
+    tap_init_amount  = 10000
+    tap = "7be017a967db77fd10ac7c891b3d6d946dea7e3e14756e2f0f9e09b9663f0d9c"
+
+    tapnonce = 0
+
+    tx_cost = 3 #.Mul(trans.GasPrice, tp.gasCost.BasicTxCost)
+
+    nonceapi = 'v1/nonce'
+    submitapi = 'v1/submittransaction'
+    balanceapi = 'v1/balance'
+
+    accounts = { tap: { "priv": "81c90dd832e18d1cf9758254327cb3135961af6688ac9c2a8c5d71f73acc5ce5", "nonce": 0, "send": [], "recv": [] } }
+
+    def random_node():
+        rnd = random.randint(0, len(setup_network.clients.pods)-1)
+        return setup_network.clients.pods[rnd]['pod_ip'], setup_network.clients.pods[rnd]['name']
+
+    def expected_balance(acc):
+        return sum([int(tx["amount"]) for tx in accounts[acc]["recv"]]) - sum(int(int(tx["amount"]) + (int(tx["gasprice"])*tx_cost)) for tx in accounts[acc]["send"])
+
+    def random_acconut():
+        pk, info = random.choice(list(accounts.keys()))
+        return pk, info
+
+    def new_account():
+        priv, pub = genkeypair()
+        strpub = bytes.hex(pub)
+        accounts[strpub] = { "priv": priv, "nonce": 0, "recv": [], "send": [] }
+        return strpub
+
+    def transfer(frm, to, amount=None, gasprice=None, gaslimit=None):
+        pod_ip, pod_name = random_node()
+        txGen = tx_generator.TxGenerator(pub=frm, pri=accounts[frm]['priv'])
+        if amount is None:
+            amount = random.randint(1, expected_balance(frm))
+        if gasprice is None:
+            gasprice = int(amount/10)
+        if gaslimit is None:
+            gaslimit = gasprice+1
+        txBytes = txGen.generate(to, accounts[frm]['nonce'], gaslimit, gasprice, amount)
+        accounts[frm]['nonce']+=1
+        data = '{"tx":'+ str(list(txBytes)) + '}'
+        out = api_call(pod_ip, data, submitapi, testconfig['namespace'])
+        print("submit transaction from {3} to {0} of {1} with gasprice {2}".format(to, amount, gasprice, frm))
+        print(data)
+        print(out)
+        if "{'value': 'ok'}" in out:
+            accounts[to]["recv"].append({ "from": txGen.publicK, "amount": amount, "gasprice": gasprice})
+            accounts[frm]["send"].append({"to": to, "amount": amount, "gasprice": gasprice})
+            return True
+        return False
+
+    def test_account(acc, init_amount=None):
+        pod_ip, pod_name = random_node()
+        #check nonce
+        data = '{"address":"'+ tap +'"}'
+        print("checking {0} nonce".format(acc))
+        out = api_call(pod_ip, data, nonceapi, testconfig['namespace'])
+        print(out)
+        if str(accounts[acc]['nonce']) in out:
+            # check balance
+            data = '{"address":"' + str(acc) + '"}'
+            print(data)
+            out = api_call(pod_ip, data, balanceapi, testconfig['namespace'])
+            print(out)
+            balance = 0
+            if init_amount is not None:
+                balance += init_amount
```

Comment:
you can set "init_amount" to 0 (default) and simply add it to balance in line 459. - Then you will not need this if statement.